app/Http/Controllers/FreeTrailController.php

```diff
@@ -0,0 +1,199 @@
+<?php
+
+namespace App\Http\Controllers;
+
+use App\Http\Controllers\Order\BaseOrderController;
+use App\Model\Common\StatusSetting;
+use App\Model\Order\Invoice;
+use App\Model\Order\InvoiceItem;
+use App\Model\Order\Order;
+use App\Model\Payment\Plan;
+use App\Model\Product\Product;
+use App\Model\Product\Subscription;
+use App\User;
+use Auth;
+use Crypt;
+use DB;
+use Illuminate\Http\Request;
+use Lang;
+
+class FreeTrailController extends Controller
+{
+    public function __construct()
+    {
+        $this->middleware('auth');
+        $invoice = new Invoice();
+        $this->invoice = $invoice;
+
+        $invoiceItem = new InvoiceItem();
+        $this->invoiceItem = $invoiceItem;
+
+        $order = new Order();
+        $this->order = $order;
+
+        $subscription = new Subscription();
+        $this->subscription = $subscription;
+    }
+
+    public function firstloginatem(Request $request)
+    {
+        if (! Auth::check()) {
+            return redirect('login')->back()->with('fails', \Lang::get('message.free-login'));
+        }
+
+        $id = $request->get('id');
+        if (Auth::user()->id == $id) {
+            $user_login = User::find($id);
+            if ($user_login->first_time_login != 0) {
+                return errorResponse(Lang::get('message.false'), 400);
+            }
+            $user = User::where('id', $id)->update(['first_time_login' => 1]);
+
+            $invoice = $this->generateFreetrailInvoice();
+
+            $invoice_items = $this->createFreetrailInvoiceItems();
+
+            $order = $this->executeFreetrailOrder();
+
+            return successResponse(Lang::get('message.succs_fre'), $order, 200);
```

Comment:
refactor the code
return successResponse(trans('message.succs_fre'), $this->executeFreetrailOrder(), 200);